modBoot/src/org/aion/Aion.java

```diff
@@ -82,6 +82,7 @@ public static void main(String args[]) {
 
         ReturnType ret = new Cli().call(args, cfg);
         if (ret != ReturnType.RUN) {
+            shutdownAvm();
```

Comment:
We can call `System.exit` from other parts of the code. Perhaps a better place for the AVM shutdown would be in the `ShutdownHook` thread defined below (around line 350).

Reply:
Ideally we register the hook first thing and it just knows whether a component has been started/loaded and how to shut it down if so, but as it stands that's one of the last things we run in the main method because we wait for everything to be started up first

Reply:
Ah, I see. Looks good to me.